octomap_mapping/octomap_server/src/zFilter.cpp

```diff
@@ -7,16 +7,21 @@
 #include <pcl_conversions/pcl_conversions.h>
 #include <visualization_msgs/Marker.h>
 #include <visualization_msgs/MarkerArray.h>
+#include <pcl/kdtree/kdtree_flann.h>
 
 // typedef pcl::PointCloud<pcl::PointXYZ> PointCloud;
 // ros::Publisher occArrayTrimmed_pub;
 float thresholdOcc = 1.0;
 int resolution = 1;
 pcl::PointCloud<pcl::PointXYZ>::Ptr cloudOccFull (new pcl::PointCloud<pcl::PointXYZ>);
+pcl::PointCloud<pcl::PointXYZ>::Ptr subset_cloud(new pcl::PointCloud<pcl::PointXYZ>);      
+
 visualization_msgs::MarkerArray zFiltered;
+visualization_msgs::MarkerArray zPruned;
+
 uint32_t shape = visualization_msgs::Marker::CUBE;
 float trimmedHeight = 3;
-
+int initstraySizeThresh = 2;
```

Comment:
Threshold for minimum number of neighbors required for a point to be a "usable" voxel